torch/_decomp/__init__.py

```diff
@@ -37,7 +40,49 @@ def clamp_min(x):
     `disable_meta` to disable this behavior.
     """
 
-    def decomposition_decorator(f):
+    def decomposition_decorator(f: Callable) -> Callable:
+        sig = inspect.signature(f)
```

Comment:
@rgommers all this block of code is the juggling with `inspect` and annotations I told you about. given a function that takes a kwargs `out=` which is a tuple of tensors and returns a namedtuple, it wraps it as a function that takes the each of the arguments as a separate tensor.

The dirty parts are:
- Accessing `out_annotations.__origin__` to check that that type is indeed a `Tuple` 
- Accessing `return_annotation._fields` to get the names from the `NamedTuple` from the return.

Reply:
Thanks @Lezcano, interesting. This is more like codegen (new annotation from old annotation) than actual use of type annotations at runtime to do anything functional, so my concern of using annotations at runtime (which is a no no) doesn't really apply. I'm not surprised that there's no clean ways in the language or in `typing`/`typing_extensions` to do this kind of thing.